contrib/docker/Dockerfile

```diff
@@ -1,60 +1,50 @@
 FROM python:3.6
 
-MAINTAINER Xiao Hanyu <hanyu.xiao@shopeemobile.com>
-
-# Add a normal user
-RUN useradd --user-group --create-home --shell /bin/bash work
-
 # Configure environment
 ENV LANG=C.UTF-8 \
-    LC_ALL=C.UTF-8 \
-    HOME=/home/work
+    LC_ALL=C.UTF-8
 
-RUN apt-get update -y
+WORKDIR /home/work/incubator-superset
 
 # Install some dependencies
 # http://airbnb.io/superset/installation.html#os-dependencies
-RUN apt-get update -y && apt-get install -y build-essential libssl-dev \
-    libffi-dev python3-dev libsasl2-dev libldap2-dev
-
-RUN apt-get install -y vim less postgresql-client redis-tools
+RUN apt-get update -y \
+    && apt-get install -y build-essential libssl-dev \
+        libffi-dev python3-dev libsasl2-dev libldap2-dev \
+        vim less postgresql-client redis-tools
 
 # Install nodejs for custom build
 # https://github.com/apache/incubator-superset/blob/master/docs/installation.rst#making-your-own-build
 # https://nodejs.org/en/download/package-manager/
-RUN curl -sL https://deb.nodesource.com/setup_8.x | bash -
-RUN apt-get install -y nodejs
-RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -; \
-    echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list; \
-    apt-get update; \
-    apt-get install -y yarn
-
-RUN mkdir $HOME/incubator-superset
+RUN curl -sL https://deb.nodesource.com/setup_8.x | bash - \
+    && apt-get install -y nodejs
```

Comment:
When I start work on superset, there is node 8.x, now maybe we can upgrade to node 10.x?

Reply:
Good idea!